layers/+lang/typescript/packages.el

```diff
@@ -2,21 +2,42 @@
 ;;
 ;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
 ;;
-;; Author: Chris Bowdon <c.bowdon@bath.edu>
-;; URL: https://github.com/syl20bnr/spacemacs
+;; Author: JAremko <w3techplaygound@gmail.com> 
+;; URL: https://github.com/JAremko/spacemacs-pr
 ;;
 ;; This file is not part of GNU Emacs.
 ;;
 ;;; License: GPLv3
 
-(setq typescript-packages '(tss))
+(setq typescript-packages '(tide))
 
-(defun typescript/init-tss ()
-  "Initialize my package"
-  (use-package tss
+(defun typescript/init-tide ()
+  (use-package tide
     :defer t
-    :mode ("\\.ts\\'" . typescript-mode)
-    :init (spacemacs/set-leader-keys-for-major-mode 'typescript-mode
-            "gg" 'tss-jump-to-definition
-            "hh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :init (progn
+            (evilified-state-evilify tide-references-mode tide-references-mode-map
+              (kbd "C-j") 'tide-find-previous-reference
+              (kbd "C-k") 'tide-find-next-reference
+              (kbd "C-l") 'tide-goto-reference)
+            (add-hook 'typescript-mode-hook
+              (lambda ()
+                (tide-setup)
+                (flycheck-mode t)
+                (setq flycheck-check-syntax-automatically '(save mode-enabled))
+                (eldoc-mode t)
+                (when (configuration-layer/package-usedp 'company) 
+                  (company-mode-on)))))
+    :config (progn
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mg" "goto")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mh" "help")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mn" "name")
+              (defun typescript-jump-to-type-definition ()
```

Comment:
Hi,

I'd like to add you to my professional network on LinkedIn.

- Robert

Confirm that you know Robert: https://www.linkedin.com/e/v2?e=0-ini67nvt-bu&t=ssuw&tracking=eml-guest-invite-cta&ek=invite_guest&sharedKey=tkWkVzCc&invitationId=6130896991670067200








You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=0-ini67nvt-bu&t=lun&midToken=AQFmrFVhGEd36Q&ek=invite_guest&loid=AQFZnuwbr_YizAAAAVRVSbWHumYNBqOETLDo7ds1zU0LWLuIpkszpZfLR7dbAHdiMFLEzM_ftgKdWYg9Ren_VYa7fcMeOlRIRcgRj70Sz26_ptn0N6SlL11mnk8Oyr36q1o34buNVfuudUj-h8hUMbQ7n0EnQORVtOU2Htx2mxzB2D3U8lUD60jWzYoK_Xt1Wc7x&eid=0-ini67nvt-bu


This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=0-ini67nvt-bu&a=customerServiceUrl&ek=invite_guest

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.